src/stream/src/executor/aggregation/agg_group.rs

```diff
@@ -30,19 +31,169 @@ use crate::common::table::state_table::StateTable;
 use crate::executor::error::StreamExecutorResult;
 use crate::executor::PkIndices;
 
+/// We assume the first state of aggregation is always `StreamingRowCountAgg`.
+const ROW_COUNT_COLUMN: usize = 0;
+
+pub trait Strategy: ChangesBuilder {}
+
+pub struct GlobalSimpleAgg;
+impl Strategy for GlobalSimpleAgg {}
+pub struct HashAgg;
+impl Strategy for HashAgg {}
```

Comment:
It seems to use `SingleGroup` and `MultiGroup` for  `Strategy` is more accurate.

Reply:
Renamed them to `AlwaysOutput` and `OnlyOutputIfHasInput`. Seems to be more accurate since one `AggGroup` struct is only for one aggregation group and can at most output one row.